File: 35_Search_Insert_Position.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        if(target>nums[-1]):
            return len(nums)
        for i in range(0,len(nums)):
            if(nums[i]==target):
                return i
            if(nums[i]>target):
                return i
        

File: 35_Search_Insert_Position.cpp
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        if (target>nums[nums.size()-1])
            return nums.size();
        int mid;
        int low=0;
        int high=nums.size()-1;
            
            while(low<=high){
            mid=low+(high-low)/2;
            if(nums[mid]==target){
                return mid;
            }
            if(nums[mid]<target){
                low=mid+1;
            }
            else{
                high=mid-1;
            }
        }
        
        if(nums[mid]>target){
            return mid;
        }
        return mid+1;
    }
};